benchmark/utils/loggers.hpp

```diff
@@ -84,27 +80,24 @@ struct JsonSummaryWriter : gko::log::ProfilerHook::SummaryWriter,
                 visit(visit, child, new_prefix);
                 exclusive -= child.elapsed;
             }
-            add_or_set_member(*object, (prefix + node.name).c_str(),
-                              exclusive.count() * 1e-9 / repetitions, *alloc);
+            (*object)[prefix + node.name] =
+                exclusive.count() * 1e-9 / repetitions;
         };
         // we don't need to annotate the total
         for (const auto& child : root.children) {
             visit(visit, child, "");
         }
-        add_or_set_member(*object, "overhead",
-                          overhead.count() * 1e-9 / repetitions, *alloc);
+        (*object)["overhead"] = overhead.count() * 1e-9 / repetitions;
     }
 
-    rapidjson::Value* object;
-    rapidjson::MemoryPoolAllocator<>* alloc;
+    json* object;
```

Comment:
maybe use `json&`? I'm not sure if we need this class to have value semantics.

Reply:
Except for rare circumstances (std:tie etc) I would consider reference members dangerous, as any assignment to the object turn into assignments into the referenced object, which is usually not intended. There is no way to rebind a reference in C++, and in this case I would say that assigning a logger should not assign its JSON object, but rather rebind it.

Reply:
Sure, I was just thinking about handling our loggers uniformly. The other loggers in this file use `json&`

Reply:
good catch, I updated them.